src/background/extraction.js

```diff
@@ -51,6 +52,11 @@ export async function handleExtractedProductData({extractedProduct}, sender) {
       tabId,
     });
     browser.browserAction.setBadgeText({text: '✚', tabId});
+    if (sendTelemetry) {
```

Comment:
What's the purpose of this? To prevent sending events when the badge isn't changing from its previous value?

If so, couldn't we just compare the old and new values instead of passing a parameter around?

Reply:
I added that second argument to `sendProductToBackground` because when we call `resend` (the messy workaround for Bug 1493470 in PR #220 ), we are also re-recording the `badge_toolbar_button` event for each call to `resend`. It is a temporary addition to the temporary code to also be removed when `resend` is removed.

Although your idea of only recording the event (for badge types of 'add' and 'price_alert') if the previous badge value is different would also work here. I'll do that instead, since that can stay even after `resend` is removed.

Reply:
Geeze this was a lot trickier than I anticipated. More soon when I resubmit the PR...

Reply:
I am still seeing duplicate `detect_price_change` events due to `resend`:

[STR]:
1. Open the [fake product page](http://www.mkelly.me/fake-product-page/) in Tab 1, say it has Price $A.
2. Add the product via the browserAction toolbar button
3. Refresh the page until a Price Alert is triggered, Price $B.
4. Immediately click on the System Notification (within 5 seconds of refreshing the page that triggered the alert)
5. This will open another fake product page in Tab 2 (at Price $C) and perform extraction before `resend` has been executed even once in Tab 1 fake product page.
6. Because the price is different between Tabs 1 and 2, the price history will look something like: [A, ... (refreshing to trigger a change), B, C, B (resend), C (resend), B (resend), C (resend)], where all the `resend` prices are duplicate events which can incorrectly trigger price alerts.

I don't think this will be a problem in any real world scenario, since it would require a product page's price to change within 5 seconds of visiting the page.

@Osmose , do you agree that this is a non-issue?

Reply:
Yeah, this should be fine.